src/pip/_internal/utils/misc.py

```diff
@@ -151,7 +152,7 @@ def rmtree_errorhandler(func: Callable[..., Any], path: str, exc_info: ExcInfo)
 def display_path(path: str) -> str:
     """Gives the display value for a given path, making it relative to cwd
     if possible."""
-    path = os.path.normcase(os.path.abspath(path))
+    path = str(Path(path).resolve())
     if path.startswith(os.getcwd() + os.path.sep):
         path = "." + path[len(os.getcwd()) :]
```

Comment:
This might not work well with the following `startswith` part. Perhaps use `relative_to()` instead?

Also please use `os.fspath()` to normalise Path objects.

Reply:
well instead of using pathlib why don't we do what it does internally ? by which i meant something like this:
```py
if WINDOWS:
        path = os.path._getfinalpathname(path).lstrip(r"\\?")
else:
        path = os.path.normcase(os.path.abspath(path))
if path.startswith(os.getcwd() + os.path.sep):
        path = "." + path[len(os.getcwd()) :]
    return path
```
how would that go?

Reply:
@uranusjr mind confirming if i should implement it in that way?